network/src/transport.rs

```diff
@@ -97,6 +100,7 @@ pub struct ConnectionMetadata {
     pub remote_peer_id: PeerId,
     pub connection_id: ConnectionId,
     pub addr: NetworkAddress,
+    pub proxy_addr: Option<NetworkAddress>,
```

Comment:
nit: it looks like this is unused, let's remove it?

Reply:
Oops yeah, should remove.

I was trying to think of ways to add both the original address (address of the proxy), and the address of the client (the one that's being encoded).  But, I think for now we can simplify it by leaving it just as the best case closest to the client address.